Assets/PatchKit Patcher/Scripts/AppUpdater/AppUpdater.cs

```diff
@@ -112,6 +105,13 @@ AppContentSummary latestVersionContentSummary
             }
         }
 
+        private long CalculateRepairCost(AppContentSummary contentSummary, IEnumerable<FileIntegrity> filesToRepair)
+        {
+            return filesToRepair
+                .Select(f => contentSummary.Files.FirstOrDefault(e => e.Path == f.FileName))
+                .Sum(f => f.Size);
```

Comment:
Will it work? Select() transforms enumerable to a new form (https://msdn.microsoft.com/en-us/library/bb548891(v=vs.110).aspx). When a file is not found, a null is returned. Sum() is dereferencing the value so it looks like NRE for me.

Reply:
When a file is not found, default value of struct `AppContentSummaryFile` is returned. There won't be any NRE here.